crates/oxc_linter/src/rules/unicorn/consistent_existence_index_check.rs

```diff
@@ -0,0 +1,493 @@
+    /// ### What it does
+    ///
+    /// Enforce consistent style for element existence checks with `indexOf()`, `lastIndexOf()`, `findIndex()`, and `findLastIndex()`
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This rule is only meant to enforce a specific style and make comparisons more clear.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    ///
+    /// ```javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index < 0) {}
+    /// ```
+    ///
+    /// ``` javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index >= 0) {}
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    ///
+    /// ```javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index === -1) {}
+    /// ```
+    ///
+    /// ``` javascript
+    /// const index = foo.indexOf('bar');
+    /// if (index !== -1) {}
+    /// ```
+    ConsistentExistenceIndexCheck,
+    correctness,
+    fix,
+);
+
+const METHOD_NAMES: [&str; 4] = ["indexOf", "lastIndexOf", "findIndex", "findLastIndex"];
+
+impl Rule for ConsistentExistenceIndexCheck {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::BinaryExpression(binary_expression) = node.kind() else {
+            return;
+        };
+
+        let left = binary_expression.left.get_inner_expression();
+        let right = binary_expression.right.get_inner_expression();
+        let operator = binary_expression.operator;
+
+        let Expression::Identifier(identifier) = left else {
+            return;
+        };
+
+        let Some(reference_id) = identifier.reference_id.get() else {
+            return;
+        };
+
+        let Some(reference_symbol_id) = ctx.symbols().get_reference(reference_id).symbol_id()
+        else {
+            return;
+        };
+
+        let declaration = ctx.symbols().get_declaration(reference_symbol_id);
+        let node = ctx.nodes().get_node(declaration);
+
+        if let AstKind::VariableDeclarator(variables_declarator) = node.kind() {
+            if variables_declarator.kind != VariableDeclarationKind::Const {
+                return;
+            }
+
+            let Some(Expression::CallExpression(call)) = &variables_declarator.init else {
+                return;
+            };
+
+            if !call.callee.is_member_expression() {
+                return;
+            }
+
+            let Some(callee_name) = call.callee_name() else {
+                return;
+            };
+
+            if !METHOD_NAMES.contains(&callee_name) {
+                return;
+            }
+
+            let replacement = get_replacement(right, &operator);
+
+            let Some(replacement) = &replacement else {
+                return;
+            };
+
+            let existence_or_non_existence =
+                if replacement.replacement_value == "-1" { "non-existence" } else { "existence" };
+
+            let label = format!(
```

Comment:
Let's move this error message logic into the diagnostic. Putting all the diagnostic messages together at the top makes it easier to see what this rule emits.